src/einsteinpy/symbolic/tensor.py

```diff
@@ -1,55 +1,327 @@
+        array = Array(matrix)
+        sym = kwargs.pop("symmetry", [[1] * array.rank()])
+        sym = tensorsymmetry(*sym)
+        symtype = TensorType(array.rank() * [metric], sym)
+        comm = kwargs.pop("comm", "general")
+        covar = tuple(kwargs.pop("covar", array.rank() * [1]))
+        if len(covar) != array.rank():
+            raise ValueError(
+                f"covariance signature {covar} does not match tensor rank {array.rank()}"
+            )
+
+        count = defaultdict(int)
+
+        def dummy_fmt_gen(idxtype):
+            # generate a generic index for the entry in ReplacementManager.
+            fmt = idxtype.dummy_fmt
+            n = count[idxtype]
+            count[idxtype] += 1
+            return fmt % n
+
+        obj = TensorHead.__new__(cls, symbol, symtype, comm=comm, **kwargs)
+        obj = AbstractTensor.__new__(cls, obj, array)
+        # resolves a bug with pretty printing.
+        # TODO: Consider renaming this class to avoid conflicts.
+        obj.__class__.__name__ = "TensorHead"
+        obj.covar = covar
+        idx_names = map(dummy_fmt_gen, obj.index_types)
+        idxs = map(Index, idx_names, obj.index_types)
+        idxs = [idx if covar[pos] > 0 else -idx for pos, idx in enumerate(idxs)]
+        ReplacementManager[obj(*idxs)] = array
+        return obj
 
-    def tensor(self):
+    def __repr__(self):
+        return self._print()
+
+    def __str__(self):
+        return str(self.__repr__())
+
+    def __call__(self, *args, **kwargs):
+        new = IndexedTensor(self, args, **kwargs)
+        return new.doit()
+
+    def covariance_transform(self, *indices):
         """
-        Returns the sympy Array
+        Return the array associated with this tensor with indices set according
+        to arguments.
+
+        Parameters
+        ----------
+        indices : TensorIndex
+            Defines the covariance and contravariance of the returned array.
+
+        Examples
+        --------
+        >>> from sympy import diag, symbols, sin
+        >>> from einsteinpy.symbolic.tensor import indices
+        >>> from einsteinpy.symbolic.metric import Metric
+        >>> t, r, th, ph = symbols('t r theta phi')
+        >>> schwarzschild = diag(1-1/r, -1/(1-1/r), -r**2, -r**2*sin(th)**2)
+        >>> g = Metric('g', [t, r, th, ph], schwarzschild)
+        >>> mu, nu = indices('mu nu', g)
+        >>> g.covariance_transform(mu, nu)
+        [[1/(1 - 1/r), 0, 0, 0], [0, 1 - 1/r, 0, 0], [0, 0, -1/r**2, 0], [0, 0, 0, -1/(r**2*sin(theta)**2)]]
 
-        Returns
-        -------
-        ~sympy.tensor.array.dense_ndim_array.ImmutableDenseNDimArray
-            Sympy Array object
-        
         """
-        return self.arr
+        array = self.as_array()
+        for pos, idx in enumerate(indices):
+            if idx.is_up ^ (self.covar[pos] > 0):
+                if idx.is_up:
+                    metric = idx.tensor_index_type.metric.as_inverse()
```

Comment:
Since this index implies that the array needs to have the components represented by this index be raised, the inverse metric is used.